doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+    >>> add(2, 2)
+    4
+    >>> add(25, 0)
+    25
+    >>> add(10, -10)
+    0
+    """
+    return num1 + num2
+
+Some standards exist about docstrings, so they are easier to read, and they can
+be exported to other formats such as html or pdf.
+
+The first conventions every Python docstring should follow are defined in
+`PEP-257 <https://www.python.org/dev/peps/pep-0257/>`_.
+
+As PEP-257 is quite open, and some other standards exist on top of it. In the
+case of pandas, the numpy docstring convention is followed. There are two main
+documents that explain this convention:
+
+- `Guide to NumPy/SciPy documentation <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_
+- `numpydoc docstring guide <http://numpydoc.readthedocs.io/en/latest/format.html>`_
+
+numpydoc is a Sphinx extension to support the numpy docstring convention.
+
+The standard uses reStructuredText (reST). reStructuredText is a markup
+language that allows encoding styles in plain text files. Documentation
+about reStructuredText can be found in:
+
+- `Sphinx reStructuredText primer <http://www.sphinx-doc.org/en/stable/rest.html>`_
+- `Quick reStructuredText reference <http://docutils.sourceforge.net/docs/user/rst/quickref.html>`_
+- `Full reStructuredText specification <http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html>`_
+
+The rest of this document will summarize all the above guides, and will
+provide additional convention specific to the pandas project.
+
+Writing a docstring
+-------------------
+
+General rules
+~~~~~~~~~~~~~
+
+Docstrings must be defined with three double-quotes. No blank lines should be
+left before or after the docstring. The text starts immediately after the
+opening quotes (not in the next line). The closing quotes have their own line
+(meaning that they are not at the end of the last sentence).
+
+**Good:**
+
+.. code-block:: python
+
+    def func():
+        """Some function.
+
+        With a good docstring.
+        """
+        foo = 1
+        bar = 2
+        return foo + bar
+
+**Bad:**
+
+.. code-block:: python
+
+    def func():
+
+        """
+        Some function.
+
+        With several mistakes in the docstring.
+        
+        It has a blank like after the signature `def func():`.
+        
+        The text 'Some function' should go in the same line as the
+        opening quotes of the docstring, not in the next line.
+        
+        There is a blank line between the docstring and the first line
+        of code `foo = 1`.
+        
+        The closing quotes should be in the next line, not in this one."""
+
+        foo = 1
+        bar = 2
+        return foo + bar
+
+Section 1: Short summary
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+The short summary is a single sentence that express what the function does in a
+concise way.
+
+The short summary must start with a verb infinitive, end with a dot, and fit in
+a single line. It needs to express what the function does without providing
+details.
```

Comment:
I think we want a length limit here, so the lines in http://pandas-docs.github.io/pandas-docs-travis/api.html don't wrap. Though it looks like we can't set a hard limit, since the width available depends on the section...

Reply:
I also don't really care about the "verb infinitive" part.

Reply:
This doesn't solve the wrapping issue you mention but I wonder if either here or in the General section we should make note that comments still need to wrap at 79 characters for PEP-8 compliance

Reply:
I checked the line length, and seems that for the rendering in autosummaries something around 60 to 75 characters is the maximum length to avoid wrapping in. The document currently states to fit in a single line, which with PEP-8 line length means 76 characters for functions and 72 for methods. Unless you have a better idea, I'll leave like it is, as I think it's much easier for people to write a single line, than to count the characters.

Reply:
Regarding the infinitive verb, it's a standard used in Python projects in investment banking. I think it helps people write concise summaries. All functions/methods do things, so starting with a verb always should make sense, and it avoids starts like "This function [...]", "Method to [...]". Being infinitive is just to standardize "Generates [...]", "Generating [...]" and "Generate [...]" to always have the same form. Not sure if any other reason was used besides the infinitive being shorter, but unless you really want to get rid of this rule, I'll keep it as it's used in investment banks.